elooplib/eventloop.cpp

```diff
@@ -63,37 +62,76 @@ int EventLoop::handleConnection(std::shared_ptr<Node> &hst)
     return -1;
 }
 
+EventLoop::nodePointerType EventLoop::waitersSearch(int fd)
+{
+    auto nodeComp = [&fd](nodeType &n)
+    {
+        return fd == n.second->getFd();
+    };
+    auto wp = find_if(begin(wait_q), end(wait_q), nodeComp);
+    return wp;
```

Comment:
find_if returns iterator, not pointer. But vector's iterator gets automatically casted to pointer